internal/pkg/service/stream/api/service/source.go

```diff
@@ -230,7 +230,17 @@ func (s *service) SourceStatisticsClear(ctx context.Context, d dependencies.Sour
 		return err
 	}
 
-	return d.StatisticsRepository().ResetStats(d.SourceKey()).Do(ctx).Err()
+	sinks, err := s.definition.Sink().List(d.SourceKey()).Do(ctx).All()
+	if err != nil {
+		return err
+	}
+
+	sinkKeys := []key.SinkKey{}
+	for _, sink := range sinks {
+		sinkKeys = append(sinkKeys, sink.SinkKey)
+	}
+
+	return d.StatisticsRepository().ResetAllSinksStats(ctx, sinkKeys)
```

Comment:
I think it makes sense. Maybe it would be an optimalization if the users are frequently asking `source.Stats, source.Reset` to calculate the stats to source and work with that numbers.